Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2537,6 +2533,18 @@ void CGContextClearRect(CGContextRef context, CGRect rect) {
 
 HRESULT __CGContext::DrawGeometry(__CGCoordinateMode coordinateMode, ID2D1Geometry* pGeometry, CGPathDrawingMode drawMode) {
     ComPtr<ID2D1Geometry> geometry(pGeometry);
+    if (coordinateMode == _kCGCoordinateModeUserSpace) {
+        // pre transform the geometry
+        ComPtr<ID2D1TransformedGeometry> transformedGeometry;
+        FAIL_FAST_IF_FAILED(
+            Factory()->CreateTransformedGeometry(geometry.Get(),
+                                                 __CGAffineTransformToD2D_F(CGContextGetUserSpaceToDeviceSpaceTransform(this)),
+                                                 &transformedGeometry));
+
+        // Draw it in device space, so transformation is not applied to the render target.
+        coordinateMode = _kCGCoordinateModeDeviceSpace;
+        geometry.Attach(transformedGeometry.Detach());
```

Comment:
note: As per D2D DrawGeo.. a transformed geometry will have a thinner line width, than an untransformed geometry + transformation.

Working on a fix in the coming iteration.